service/status-report/main.go

```diff
@@ -156,28 +154,16 @@ func setClusterInformation(ctx context.Context, cl client.Client, status interfa
 	}
 	status.SetClusterID(string(clusterID))
 
-	clusterConfig := &corev1.ConfigMap{}
-	clusterConfig.Name = clusterConfigName
-	clusterConfig.Namespace = clusterConfigNamespace
-
-	if err := cl.Get(ctx, client.ObjectKeyFromObject(clusterConfig), clusterConfig); err != nil {
-		klog.Warningf("Failed to get clusterConfig %q/%q: %v", clusterConfig.Namespace, clusterConfig.Name, err)
+	clusterDNS := &configv1.DNS{}
+	clusterDNS.Name = clusterDNSName
+	if err := cl.Get(ctx, client.ObjectKeyFromObject(clusterDNS), clusterDNS); err != nil {
+		klog.Warningf("Failed to get clusterDNS %q: %v", clusterDNS.Name, err)
 	}
 
-	clusterMetadataYAML := clusterConfig.Data["install-config"]
-	clusterMetadata := struct {
-		Metadata struct {
-			Name string `yaml:"name"`
-		} `yaml:"metadata"`
-	}{}
-	err := yaml.Unmarshal([]byte(clusterMetadataYAML), &clusterMetadata)
-	if err != nil {
-		klog.Warningf("Failed to unmarshal cluster metadata, %v", err)
-	}
-	if len(clusterMetadata.Metadata.Name) == 0 {
-		klog.Warningf("Cluster name is empty, %v", err)
+	if len(clusterDNS.Spec.BaseDomain) == 0 {
+		klog.Warningf("Cluster Base Domain is empty.")
 	}
-	status.SetClusterName(clusterMetadata.Metadata.Name)
+	status.SetClusterName(clusterDNS.Spec.BaseDomain)
```

Comment:
@rchikatw does the BaseDomain string includes the cluster's domain name or only the ORG's domain name (the suffix)?

Reply:
I am not 100% sure about this, the Base domain is the API hostname From what I understood, @leelavg Can you correct if I am wrong.

Reply:
Checked with cnv team: 
in the DNS CR's `.spec.baseDomain`, it's the cluster name + org domain name
in `install-config.yaml`, it is separated:
`.baseDomain` is only the org's domain name
`.metadata.name` is the cluster name
however, the second option is not available in HCP clusters

Reply:
Ok, so we are good. I have advocated using the full name anyways. This will enable whoever consumes the name to make their own decision on which part is important to them.